mesonbuild/coredata.py

```diff
@@ -719,19 +736,20 @@ def replace(v):
         opt.set_value(value)
 
         if key.name == 'buildtype':
+            assert isinstance(value, str), 'for mypy'
             self._set_others_from_buildtype(value)
         elif key.name in {'wrap_mode', 'force_fallback_for'}:
             # We could have the system dependency cached for a dependency that
             # is now forced to use subproject fallback. We probably could have
             # more fine grained cache invalidation, but better be safe.
             self.clear_deps_cache()
 
-    def clear_deps_cache(self):
+    def clear_deps_cache(self) -> None:
         self.deps.host.clear()
         self.deps.build.clear()
 
-    def get_nondefault_buildtype_args(self):
-        result = []
+    def get_nondefault_buildtype_args(self) -> T.List[T.Union[T.Tuple[Literal['optimization'], str, str], T.Tuple[Literal['debug'], bool, bool]]]:
+        result: T.List[T.Union[T.Tuple[Literal['optimization'], str, str], T.Tuple[Literal['debug'], bool, bool]]] = []
```

Comment:
This is long enough and difficult enough to read, *just* because of the annotation, that I would like to see it defined as a readable type in T.TYPE_CHECKING and then reused here, even if it was only used once. :D But lucky us, we can use it twice.

Reply:
Done, but feel free to bikeshed the name